src/lib/structures/base/Store.js

```diff
@@ -133,13 +133,10 @@ class Store extends Collection {
 	 * Sets up a piece in our store.
 	 * @since 0.0.1
 	 * @param {Piece} piece The piece we are setting up
-	 * @returns {?Piece}
+	 * @returns {Piece}
 	 */
 	set(piece) {
-		if (!(piece instanceof this.holds)) {
-			this.client.emit('error', `Only ${this} may be stored in this Store.`);
-			return null;
-		}
+		if (!(piece instanceof this.holds)) throw new Error(`Only ${this} may be stored in this Store.`);
```

Comment:
Is TypeError better than just Error?

Reply:
TypeError would be a more accurate error to use, yes